rust/arrow/src/array/builder.rs

```diff
@@ -1214,11 +1214,12 @@ impl DecimalBuilder {
         }
     }
 
-    /// Appends a byte slice into the builder.
+    /// Append i128 to into the builder. For DecimalNType please use append_value.
+    /// This method is useful when you are reading a data and doesnt require to cast value to DecimalNType
     ///
     /// Automatically calls the `append` method to delimit the slice appended in as a
     /// distinct array element.
-    pub fn append_value(&mut self, value: i128) -> Result<()> {
+    pub fn append_value_i128(&mut self, value: i128) -> Result<()> {
```

Comment:
What about something like 

```suggestion
    pub fn append_value(&mut self, value: impl Into<ArrowDecimalType>) -> Result<()> {
      let value: ArrowDecimalType = value.into();
```

And then as long as you implement `impl From<i128>` for Decimal128Type` it should be possible to call `append_value` with either a `Decimal128Type` or an `i128`